dali/kernels/imgproc/color_manipulation/hsv_cpu_test.cc

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+#include "dali/kernels/imgproc/color_manipulation/hsv_cpu.h"
+#include "dali/kernels/imgproc/color_manipulation/color_manipulation_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace hsv {
+namespace test {
+
+namespace {
+
+static constexpr int ndims = 3;
+
+void fill_roi(Box<2, int> &roi) {
+    roi = {{1, 2},
+           {5, 7}};
+}
+
+
+template<class GtestTypeParam>
+using TheKernel = HsvCpu<typename GtestTypeParam::Out, typename GtestTypeParam::In>;
+
+}  // namespace
+
+
+template<class InputOutputTypes>
+class HsvCpuTest : public ::testing::Test {
+ protected:
+    HsvCpuTest() {
+        input_.resize(dali::volume(shape_));
+        ref_output_.resize(dali::volume(shape_));
+    }
+
+
+    void SetUp() final {
+        std::mt19937_64 rng;
+        UniformRandomFill(input_, rng, 0., 10.);
+        calc_output<typename InputOutputTypes::Out>();
+        ref_out_tv_ = make_tensor_cpu(ref_output_.data(), shape_);
+    }
+
+
+    std::vector<typename InputOutputTypes::In> input_;
+    std::vector<typename InputOutputTypes::Out> ref_output_;
+    OutTensorCPU<typename InputOutputTypes::Out, ndims> ref_out_tv_;
+    TensorShape<3> shape_ = {23, 45, ndims};
+    float hue_ = 2.8f;
+    float saturation_ = 1.9f;
+    float value_ = 0.3f;
+
+
+    template<typename OutputType>
+    std::enable_if_t<std::is_integral<OutputType>::value> calc_output() {
+        for (size_t i = 0; i < input_.size(); i += ndims) {
+            ref_output_[i] = std::round(input_[i] + hue_);
+            ref_output_[i + 1] = std::round(input_[i + 1] * saturation_);
+            ref_output_[i + 2] = std::round(input_[i + 2] * value_);
+        }
+    }
+
+
+    template<typename OutputType>
+    std::enable_if_t<!std::is_integral<OutputType>::value> calc_output() {
+        for (size_t i = 0; i < input_.size(); i += ndims) {
+            ref_output_[i] = input_[i + 0] + hue_;
```

Comment:
```suggestion
            ref_output_[i] = input_[i] + hue_;
```
or add the `+ 0` in the integral case

Reply:
Done